packages/agreement-platformstate-writer/src/consumerServiceV2.ts

```diff
@@ -1,27 +1,279 @@
 import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
-import { AgreementEventEnvelopeV2 } from "pagopa-interop-models";
+import {
+  Agreement,
+  AgreementEventEnvelopeV2,
+  AgreementV2,
+  fromAgreementV2,
+  genericInternalError,
+  makeGSIPKConsumerIdEServiceId,
+  makeGSIPKEServiceIdDescriptorId,
+  makePlatformStatesAgreementPK,
+  makePlatformStatesEServiceDescriptorPK,
+  PlatformStatesAgreementEntry,
+  PlatformStatesCatalogEntry,
+} from "pagopa-interop-models";
 import { match } from "ts-pattern";
+import {
+  agreementStateToItemState,
+  deleteAgreementEntry,
+  readAgreementEntry,
+  readCatalogEntry,
+  updateAgreementStateInPlatformStatesEntry,
+  updateAgreementStateOnTokenStates,
+  updateAgreementStateAndDescriptorInfoOnTokenStates,
+  writeAgreementEntry,
+  isLatestAgreement,
+} from "./utils.js";
 
 export async function handleMessageV2(
   message: AgreementEventEnvelopeV2,
-  _dynamoDBClient: DynamoDBClient
+  dynamoDBClient: DynamoDBClient
 ): Promise<void> {
   await match(message)
+    .with({ type: "AgreementActivated" }, async (msg) => {
```

Comment:
one more point: can we add the check `isLatestAgreement` also on activation?
(it wasn't in the doc, adding it now)

Reply:
Done